src/components/ConnectWallet.tsx

```diff
@@ -15,19 +16,33 @@ interface ConnectWalletProps {
 }
 
 const ConnectWallet = ({ currentTheme }: ConnectWalletProps) => {
-  const userSession = useAppSelector(selectUsereSessionState);
+  const [finalStatus, setFinalStatus] = useState<string>('Viewer');
+  const userSession = useAppSelector(selectUserSessionState);
   const dispatch = useAppDispatch();
 
   const currentRole = useAppSelector(selectCurrentUserRole);
   const location = useLocation();
 
   const controlAccessRoutes = () => {
     if (location.pathname !== '/') {
-      if (location.pathname.substring(1).toLowerCase() !== currentRole.toLowerCase()) {
+      if (location.pathname.substring(1)?.toLowerCase() !== currentRole.toLowerCase()) {
         console.log('Seems like you got lost, click here to go back to the main page');
       }
     }
   };
+  useEffect(() => {
+    const fetchStatus = async () => {
+      const args = userSession.loadUserData().profile.stxAddress.testnet;
```

Comment:
this should take the address based on the network from const file.

mainnet => .mainnet
testnet/devnet => .testnet

Reply:
it is not changed in a future opened pr. can be done in a future commit.